web/src/client/js/libs/currentResource.js

```diff
@@ -1,38 +1,43 @@
 // Get the current resource based on the url
-// Takes react-router's match obj as arg
+// Takes react-router's history path as arg, and produces params using matchPath
+// This is so components _outside_ of a Route still work
 //
 // useDataService compatible function
+import { matchPath } from 'react-router-dom'
 import dataMapper from './dataMapper'
 
-const urlToResource = [
-  {
-    regex: new RegExp('.*project/.*'),
-    func: dataMapper.project.id,
-    idName: 'projectId'
-  },
-  // Catch-all for empty current resource
-  // Must be last in array!
-  {
-    regex: new RegExp('.*'),
-    func: () => {
-      return {
-        fetchAction: () => {},
-        getLoadingStatusFromState: () => {},
-        getDataFromState: () => {}
-      }
-    },
-    idName: null
+function returnNull() {
+  return {
+    fetchAction: () => {},
+    getLoadingStatusFromState: () => {},
```

Comment:
```suggestion
    getLoadingStatusFromState: () => false,
```
The current implementation has a bug where once the resource is set, if it moves to an undefined resource the previous resource will never be unset. Returning false here tells useDataService the resource is loaded and will then set the appropriate (undefined in this case) data and the component using the hook will update accordingly

Reply:
Nice